Implement night sight hardware effect with OpenGL

inguin opened the pull request:
When the night sight hardware is enabled, let the shader draw everything at full light level and in black and white.

There are a few noticeable differences compared to the software renderer:
- The original effect is more blueish. We could approximate the parameters experimentally, or try to recalculate them from the shading table.
- Software night sight suppresses all palette effects (blinking lights on textures or objects, etc.).
- With software night sight, stars become invisible through structured windows (like the grove ceilings). I think that's a bug, because simple windows continue to show stars.

Fixes #135

inguin commented:
It just occurred to me that the grayscale conversion is now happening twice: Once when the scene is rendered into the off-screen texture and once when blitting to the screen. Makes no difference as long as the red, green and blue components are equal, but it will make a difference if we try to reproduce the blueish look.

It woule be more correct to render everything in color and at full brightness, then convert to finished scene to grayscale.

Interrupt commented:
Yeah, I agree. This would be better as a post process effect since we have to draw the screen one more time anyway.

inguin commented:
Updated the implementation to render everything at full brightness and in color, then convert to grayscale in the end. I added a bit of a blue tint to the gray levels; it doesn't exactly meet the original colors, but it comes pretty close.

I'm not using the colorShader because I was planning to eventually change that one to work with a simple color attribute instead of a 1x1-pixel dummy texture. Should be way more efficient than the current hack.

PS: Tried to log into Discord some weeks ago, but they wouldn't let me in again. Each time they made me solve a dozen annoying captchas, then they said I was logging in from a new location and should check my mail. The mail arrived a few hours later, but by then the confirmation link had long timed out. I'll try it again now.
